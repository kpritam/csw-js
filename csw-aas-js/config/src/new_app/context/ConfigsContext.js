import React, {useState} from "react";

// const e = () => {
// }

const ConfigsContext = React.createContext([]);

export const ConfigsContextProvider = (props) => {
  const [items, setItems] = useState([]);

  const addItem = (item) => {
    setItems(items.concat(item))
  };

  return <ConfigsContext.Provider value={{items, addItem, setItems}}>
    {props.children}
  </ConfigsContext.Provider>
};
export default ConfigsContext
