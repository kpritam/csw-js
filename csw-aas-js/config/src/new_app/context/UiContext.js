import React, {useState} from "react";

const e = () => {
};

export const UiContext = React.createContext({
  addModalOpen: false
});

export const UiContextProvider = (props) => {
  const [uiState, setUIState] = useState({
    addModalOpen: false
  });

  const openAddModal = () => {
    setUIState({...uiState, addModalOpen: true});
  };

  const closeAddModal = () => {
    setUIState({...uiState, addModalOpen: false});
  };

  return <UiContext.Provider value={{addModalOpen: uiState.addModalOpen, openAddModal, closeAddModal}}>
    {props.children}
  </UiContext.Provider>
};

export default UiContext
