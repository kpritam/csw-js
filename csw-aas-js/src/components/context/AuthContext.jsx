import React from 'react'

/**
 * Default state for AuthContextProvider
 * @type {{auth: undefined, login: (function(): boolean), logout: (function(): boolean)}}
 */
const AuthContextDefaultState = {
  auth: undefined,
  // fixme: Why do we need login and logout, types correct?
  login: () => true,
  logout: () => true,
}

const AuthContext = React.createContext(AuthContextDefaultState)
const { Provider, Consumer } = AuthContext

// todo: AuthContext is exported to support scala.js, see if only exporting Consumer works
export { AuthContext, Provider, Consumer, AuthContextDefaultState }
