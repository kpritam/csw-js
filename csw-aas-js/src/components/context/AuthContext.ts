import React from 'react'
import { Auth } from '../Auth'

export interface AuthContextType {
  auth: Auth | null
  login: () => void
  logout: () => void
}

/**
 * Default state for AuthContextProvider
 * @type {{auth: undefined, login: (function(): boolean), logout: (function(): boolean)}}
 */
const AuthContextDefaultState = {
  auth: null,
  login: () => {},
  logout: () => {},
}

const AuthContext = React.createContext<AuthContextType>(
  AuthContextDefaultState,
)
const { Provider, Consumer } = AuthContext

// todo: AuthContext is exported to support scala.js, see if only exporting Consumer works
export { AuthContext, Provider, Consumer, AuthContextDefaultState }
