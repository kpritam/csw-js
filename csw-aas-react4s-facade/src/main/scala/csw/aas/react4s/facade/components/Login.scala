package csw.aas.react4s.facade.components

import com.github.ahnfelt.react4s.{JsComponent, JsComponentConstructor}
import csw.aas.react4s.facade.components.internal.JsAuthComponents

object Login {
  def apply(): JsComponentConstructor = RawComponent()

  private object RawComponent extends JsComponent(JsAuthComponents.Login)
}
