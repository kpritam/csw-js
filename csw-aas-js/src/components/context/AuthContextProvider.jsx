import React, { useState, useEffect } from 'react'
import { Provider } from './AuthContext'
import PropTypes from 'prop-types'
import { Auth } from '../Auth'

/**
 * React component which is wrapper over provider of react context api.
 * Responsible for instantiating keycloak and provide context value to consumers
 * props -
 * config json specific to UI application e.g. realm and clientId
 * children - react component or html element which can have consumer to access
 * context provided
 */
const AuthContextProvider = props => {
  const [auth, setauth] = useState(undefined)

  useEffect(() => {
    loginWithoutRedirect()
  }, [])

  /**
   * Instantiate keycloak and sets TMTAuthStore instance in state. This state can be provided
   * as a context
   */
  const instantiateAAS = async (url, redirect) => {
    const { keycloak, authenticated } = await Auth.authenticate(
      props.config,
      url,
      redirect,
    )
    authenticated
      .success(() => {
        const _auth = Auth.from(keycloak)
        setauth(_auth)
      })
      .error(() => {
        setauth(null)
      })
  }

  /**
   * Resolves AAS server and instantiate keycloak in check-sso mode
   */
  const loginWithoutRedirect = async () => {
    const url = await Auth.getAASUrl()
    await instantiateAAS({ url: url }, false)
  }

  /**
   * Resolves AAS server and instantiate keycloak in login-required mode
   */
  const login = async () => {
    const url = await Auth.getAASUrl()
    await instantiateAAS({ url: url }, true)
  }

  const logout = async () => {
    const logoutPromise = await auth.logout()
    logoutPromise.success(() => {
      setauth(null)
    })
  }

  return (
    <Provider value={{ auth: auth, login: login, logout: logout }}>
      {props.children}
    </Provider>
  )
}

AuthContextProvider.propTypes = {
  config: PropTypes.object,
  children: PropTypes.node,
}

export default AuthContextProvider
