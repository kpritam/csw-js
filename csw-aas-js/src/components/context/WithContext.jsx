import React from 'react'
import { Consumer } from './AuthContext'

/**
 * Higher order function which returns Component with additional props e.g auth
 * login and logout used through Consumer of context
 * @param Component react component with some props
 */
const WithContext = Component => {
  return props => {
    return (
      <Consumer>
        {({ auth, login, logout }) => {
          return (
            <Component {...props} auth={auth} login={login} logout={logout} />
          )
        }}
      </Consumer>
    )
  }
}

export default WithContext
