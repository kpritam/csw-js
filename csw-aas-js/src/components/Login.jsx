import React from 'react'
import PropTypes from 'prop-types'
import WithContext from './context/WithContext'

/**
 * React component which renders Login button.
 */
export class Login extends React.Component {
  render() {
    return (
      <button
        id='aas-login'
        onClick={
          // Call to this.props.login method is responsible for resolving and instantiating AAS server
           this.props.login
        }>
        Login
      </button>
    )
  }
}

Login.propTypes = {
  // WithContext utility provides login method as a prop.
  login: PropTypes.func,
}

export default WithContext(Login)
