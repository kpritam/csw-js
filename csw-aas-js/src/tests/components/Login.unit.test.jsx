import React from 'react'
import Enzyme, { mount } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import Login from '../../components/Login'
import renderer from 'react-test-renderer'
import { AuthContext } from '../../components/context/AuthContext'

// DEOPSCSW-630 - Javascript adapter for AAS
// DEOPSCSW-631 - React layer for javascript adapter for AAS
describe('<Login />', () => {
  Enzyme.configure({ adapter: new Adapter() })

  it('should call login', async () => {
    const authContext = { login: jest.fn() }
    const wrapper = await mount(
      <AuthContext.Provider value={{ ...authContext }}>
        <Login />
      </AuthContext.Provider>,
    )

    wrapper.find('button').simulate('click')

    expect(authContext.login).toHaveBeenCalled()

    wrapper.unmount()
  })

  it('should render login', () => {
    const props = {
      login: jest.fn(),
    }
    const login = renderer.create(<Login {...props} />).toJSON()
    expect(login).toMatchSnapshot()
  })
})
