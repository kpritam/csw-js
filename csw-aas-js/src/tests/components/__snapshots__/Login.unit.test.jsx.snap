// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> should render login 1`] = `
<button
  id="aas-login"
  onClick={[MockFunction]}
>
  Login
</button>
`;
